core/blockchain.go

```diff
@@ -1447,20 +1437,14 @@ func (bc *BlockChain) writeKnownBlock(block *types.Block) error {
 // writeBlockWithState writes block, metadata and corresponding state data to the
 // database.
 func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.Receipt, statedb *state.StateDB) error {
-	// Calculate the total difficulty of the block
-	ptd := bc.GetTd(block.ParentHash(), block.NumberU64()-1)
```

Comment:
Note, this is a backwards-incompatible change. Once blocks without TD have been written, users cannot switch back to an earlier version, right? 
And if so, perhaps we need to version-bump after merging this PR ?

Reply:
Also bump the db version number?

Reply:
If so, i would like to include my freezer pr as well. It also needs a db version bump

Reply:
It is 100% a one directional move, we should bump both.